hash.ts

```diff
@@ -10,7 +10,148 @@ export type Description = Record<string, string | string[]> & {
   eNS?: Record<string, Record<string, string>>;
 };
 
-const referenceLookups: Record<string, (e: Element) => Element[]> = {};
+/* currently only works if doName is set, GSSE support is missing */
+function findFCDAReferences(e: Element): Element[] {
+  const ldInst = e.getAttribute('ldInst');
+  const prefix = e.getAttribute('prefix');
+  const lnClass = e.getAttribute('lnClass');
+  const lnInst = e.getAttribute('lnInst');
+  const doName = e.getAttribute('doName');
+  const daName = e.getAttribute('daName');
+  const fc = e.getAttribute('fc');
+
+  if (!doName) {
+    return [];
+  }
+
+  const references = [] as Element[];
+
+  const server = e.closest('Server');
+  if (!server) {
+    return [];
+  }
+
+  const ln = server.querySelector(
+    `:scope>LDevice[inst="${ldInst}"]>LN0[prefix="${prefix}"][lnClass="${lnClass}"],
+     :scope>LDevice[inst="${ldInst}"]> LN[prefix="${prefix}"][lnClass="${lnClass}"][inst="${lnInst}"]`,
```

Comment:
Because (regrettably) so many FCDA attributes are optional and because the standard is unclear we have to deal with another special case.

In Ed 2.1 for `prefix`: "Prefix identifying together with lnInst and lnClass the LN where the DO resides; optional, default value is the empty string". Often we interpret this as `prefix=""` or `prefix` missing entirely as equivalent.

Prefix is optional within `tLN` with a default of `""` so I think this is a legitimate case we should consider and for that I am sorry :wink: (and the situation is worse with `ExtRef` elements!)